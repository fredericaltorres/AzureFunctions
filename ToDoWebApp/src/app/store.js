import { createStore, applyMiddleware, compose } from "redux";
import { connectRouter, routerMiddleware } from "connected-react-router";
import thunk from "redux-thunk";
import createHistory from "history/createBrowserHistory";
import rootReducer from "app/rootReducer";
import { createToDoFetchAsyncAction } from "../entities/todo/todoActionCreator";

import tracer from "../common/Tracer";

import actionTracerMiddleware from "../middleware/actionTracerMiddleware";

export const history = createHistory();

const initialState 	= {};
const enhancers 	= [];
const middleware 	= [thunk, routerMiddleware(history), actionTracerMiddleware];

tracer.log(`Store Init - NODE_ENV:${process.env.NODE_ENV }`);

if (process.env.NODE_ENV === "development") {

	const devToolsExtension = window.__REDUX_DEVTOOLS_EXTENSION__;
	if (typeof devToolsExtension === "function") {
		enhancers.push(devToolsExtension());
	}
}

const composedEnhancers = compose(
	applyMiddleware(...middleware),
	...enhancers
);

const store = createStore(connectRouter(history)(rootReducer), initialState, composedEnhancers);

console.log(`Loading data . . .`);
store.dispatch(createToDoFetchAsyncAction()).then((response) => {
 	console.log(`Done`);
 });

export default store;
