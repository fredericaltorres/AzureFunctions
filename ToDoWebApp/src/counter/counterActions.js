import * as actionTypes from './counterActionTypes';

export const incrementAction = () => {  
    const m = {
        type: actionTypes.INCREMENT
    } 
    console.log(`counterActions.incrementAction() return ${JSON.stringify(m)}`);
    return m;
};
export const decrementAction = () => {  
    const m = {
        type: actionTypes.DECREMENT
    } 
    console.log(`counterActions.decrementAction() return ${JSON.stringify(m)}`);
    return m;
};

