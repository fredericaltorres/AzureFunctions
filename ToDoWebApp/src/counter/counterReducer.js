import * as counterActionTypes from './counterActionTypes';

const counterReducer = (state = { value: 0 }, action) => {

    const logOn = !action.type.startsWith('@@');
    let newState = null;

    switch (action.type) {
        case counterActionTypes.INCREMENT:
            newState = {
                ...state,
                value: state.value + 1
            };
            break;
        case counterActionTypes.DECREMENT:
            newState = {
                ...state,
                value: state.value - 1
            };
            break;
        default:
            if(logOn) console.log(`counterReducer action.type:${action.type} Ignored`);
            newState = state;
            return newState;
            break;
    }
    if(logOn) console.log(`counterReducer old-state:${JSON.stringify(state)} action:${JSON.stringify(action)}`);
    if(logOn) console.log(`counterReducer new-state:${JSON.stringify(newState)} `);
    return newState;
}

export default counterReducer;