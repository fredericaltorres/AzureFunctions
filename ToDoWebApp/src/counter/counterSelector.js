/*
    jest expect doc: https://facebook.github.io/jest/docs/en/expect.html
 */
export function counterGetCounterSelector(state) {
    
    if(state === undefined) throw new Error(`state parameter cannot be undefined`);
    if(state.counter === undefined) throw new Error(`state.counter cannot be undefined`);
    if(state.counter.value === undefined) throw new Error(`state.counter.value cannot be undefined`);
    
    return state.counter.value;
}
