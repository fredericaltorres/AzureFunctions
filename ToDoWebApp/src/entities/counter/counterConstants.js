import appConstants from "common/appConstants";

export const prefix = "counter";
export const prefixPath = `${appConstants.entityPrefix}.${prefix}`;
