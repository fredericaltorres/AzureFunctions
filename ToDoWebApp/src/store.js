import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import thunkMiddleware  from 'redux-thunk';

import counterReducer   from './counter/counterReducer';
import userReducer      from './user/userReducer';
import weatherReducer   from './weather/weatherReducer';

const rootReducer = combineReducers({
    counter : counterReducer,
    user    : userReducer,
    weather : weatherReducer,
});

const composeEnhancers  = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const middleware        = [thunkMiddleware];
const store             = createStore(rootReducer, {}, composeEnhancers(applyMiddleware(...middleware)));

store.subscribe(function(){
    //console.log('STORE SUBSCRIBED ');
});

//store.dispatch({type:'WEATHER_REQUEST_STARTED'});

export default store;
