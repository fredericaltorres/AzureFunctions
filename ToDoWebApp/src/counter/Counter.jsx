import React from "react";
import { connect } from "react-redux";
import * as counterActions from "./counterActions";
import * as counterSelector from './counterSelector';
import store from '../store';

export const CounterPlainComponent = props => {
	const { counter, increment, decrement } = props;
	console.log(`Component Counter.render()`);
	return (
		<div style={{ margin: "40px" }}>
			<span style={{ marginRight: "20px" }}>Value: {counter}</span>
			<button onClick={increment}>+</button>
			<button onClick={decrement}>-</button>
		</div>
	);
};


const defaultState = { counter: { value : 0 } };

export const mapStateToProps = function(state = defaultState) {
	const a = {
		counter: counterSelector.counterGetCounterSelector(state)
	};
	console.log(`Component Counter.mapStateToProps() state:${JSON.stringify(state)}, return:${JSON.stringify(a)}`);
	return a;
}

const mapDispatchToProps = {
    increment: counterActions.incrementAction,
    decrement: counterActions.decrementAction
};

console.log(`Component Counter connect, connector`);

const connector = connect(mapStateToProps, mapDispatchToProps);
const connectedComponent = connector(CounterPlainComponent);
export default connectedComponent;
