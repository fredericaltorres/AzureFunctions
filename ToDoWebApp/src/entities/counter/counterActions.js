import { createAction } from "redux-actions";
import { createAsyncAction } from "@emoney/redux-utils";
import { prefix } from "./counterConstants";
import { incrementProcessor, decrementProcessor } from "./counterProcessors";

export const INCREMENT_ACTION_TYPE = `${prefix}/INCREMENT`;
export const DECREMENT_ACTION_TYPE = `${prefix}/DECREMENT`;
export const INCREMENT_ASYNC_ACTION_TYPE = `${prefix}/INCREMENT_ASYNC`;
export const DECREMENT_ASYNC_ACTION_TYPE = `${prefix}/DECREMENT_ASYNC`;

export const incrementAction = createAction(INCREMENT_ACTION_TYPE);

export const decrementAction = createAction(DECREMENT_ACTION_TYPE);

export const incrementAsyncAction = createAsyncAction(
	INCREMENT_ASYNC_ACTION_TYPE,
	incrementProcessor
);
export const decrementAsyncAction = createAsyncAction(
	DECREMENT_ASYNC_ACTION_TYPE,
	decrementProcessor
);
