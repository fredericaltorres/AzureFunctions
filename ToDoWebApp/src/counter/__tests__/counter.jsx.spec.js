import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

// Import the component to test as a plain component
import { CounterPlainComponent } from '../counter';
// Import the component to test wrapped in Provider and need store thanks to redux
import Counter from '../counter'; 
// react-redux Provider require a store
import store from '../../store';

describe(`The Counter component `, () => {

    it(`render plain`, () => {

        const tree = renderer.create( 
            <CounterPlainComponent counter="1234" /> 
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});