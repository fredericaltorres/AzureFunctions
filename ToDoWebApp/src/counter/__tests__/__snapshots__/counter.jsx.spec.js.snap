// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Counter component  render plain 1`] = `
<div
  style={
    Object {
      "margin": "40px",
    }
  }
>
  <span
    style={
      Object {
        "marginRight": "20px",
      }
    }
  >
    Value: 
    1234
  </span>
  <button
    onClick={undefined}
  >
    +
  </button>
  <button
    onClick={undefined}
  >
    -
  </button>
</div>
`;
